app/src/evm/evm_eip191.c

```diff
@@ -0,0 +1,144 @@
+/*******************************************************************************
+ *  (c) 2018 - 2024 Zondax AG
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ********************************************************************************/
+#include "evm_eip191.h"
+
+#include "actions.h"
+#include "apdu_handler_evm.h"
+#include "app_main.h"
+#include "app_mode.h"
+#include "coin_evm.h"
+#include "zxformat.h"
+#include "zxmacros.h"
+
+#if defined(LEDGER_SPECIFIC)
+#include "cx.h"
+#else
+#define CX_SHA256_SIZE 32
+#define CX_RIPEMD160_SIZE 20
+#endif
+
+static const char SIGN_MAGIC[] =
+    "\x19"
+    "Ethereum Signed Message:\n";
+
+static void eip191_uint32_to_str(uint32_t num, char *str) {
```

Comment:
No need to implement it from scratch, we have functions in [ledger-zxlib](https://github.com/Zondax/ledger-zxlib/blob/main/include/zxformat.h#L27) for that conversion



so `uint32_to_str` generated by the macro and full tested and used in other apps